src/test/java/tr/com/srdc/cda2fhir/AllergyConcernActTest.java

```diff
@@ -0,0 +1,391 @@
+import org.openhealthtools.mdht.uml.cda.consol.impl.AllergyProblemActImpl;
+import org.openhealthtools.mdht.uml.cda.consol.impl.AllergyStatusObservationImpl;
+import org.openhealthtools.mdht.uml.cda.util.CDAUtil;
+import org.openhealthtools.mdht.uml.cda.consol.impl.ConsolFactoryImpl;
+import org.openhealthtools.mdht.uml.cda.consol.impl.ReactionObservationImpl;
+import org.openhealthtools.mdht.uml.cda.impl.CDAFactoryImpl;
+import org.openhealthtools.mdht.uml.cda.impl.EntryRelationshipImpl;
+import org.openhealthtools.mdht.uml.cda.impl.Participant2Impl;
+import org.openhealthtools.mdht.uml.cda.impl.PlayingEntityImpl;
+import org.openhealthtools.mdht.uml.hl7.datatypes.CD;
+import org.openhealthtools.mdht.uml.hl7.datatypes.CE;
+import org.openhealthtools.mdht.uml.hl7.datatypes.CS;
+import org.openhealthtools.mdht.uml.hl7.datatypes.DatatypesFactory;
+import org.openhealthtools.mdht.uml.hl7.datatypes.II;
+import org.openhealthtools.mdht.uml.hl7.datatypes.IVL_TS;
+import org.openhealthtools.mdht.uml.hl7.datatypes.IVXB_TS;
+import org.openhealthtools.mdht.uml.hl7.datatypes.impl.DatatypesFactoryImpl;
+import org.openhealthtools.mdht.uml.hl7.vocab.EntityClassRoot;
+import org.openhealthtools.mdht.uml.hl7.vocab.ParticipationType;
+import org.openhealthtools.mdht.uml.hl7.vocab.RoleClassRoot;
+import org.openhealthtools.mdht.uml.hl7.vocab.x_ActRelationshipEntryRelationship;
+
+import com.bazaarvoice.jolt.JsonUtils;
+
+import tr.com.srdc.cda2fhir.transform.ResourceTransformerImpl;
+
+public class AllergyConcernActTest {
+	private static final ResourceTransformerImpl rt = new ResourceTransformerImpl();
+	
+	private static ConsolFactoryImpl cdaObjFactory;
+	private static DatatypesFactory cdaTypeFactory;
+	private static CDAFactoryImpl cdaFactory;
+	
+	private static Map<String, String> cdaProblemStatusCodeToName = new HashMap<String, String>();
+	private static Map<String, String> cdaAllergyIntolaranceTypeCodeToName = new HashMap<String, String>();
+	
+	private static Map<String, Object> clinicalStatusMap = JsonUtils.filepathToMap("src/test/resources/jolt/value-maps/AllergyIntoleranceClinicalStatus.json");
+	private static Map<String, Object> verificationStatusMap = JsonUtils.filepathToMap("src/test/resources/jolt/value-maps/AllergyIntoleranceVerificationStatus.json");
+	private static Map<String, Object> categoryMap = JsonUtils.filepathToMap("src/test/resources/jolt/value-maps/AllergyIntoleranceCategory.json");
+		
+	@BeforeClass
+	public static void init() {
+		CDAUtil.loadPackages();
+		
+		cdaObjFactory = (ConsolFactoryImpl) ConsolFactoryImpl.init();
+		cdaTypeFactory = DatatypesFactoryImpl.init();		
+		cdaFactory = (CDAFactoryImpl) CDAFactoryImpl.init();
+		
+		cdaProblemStatusCodeToName.put("55561003", "Active");
+		cdaProblemStatusCodeToName.put("73425007", "Inactive");
+		cdaProblemStatusCodeToName.put("413322009", "Resolved");
+		
+		cdaAllergyIntolaranceTypeCodeToName.put("419199007", "Allergy to substance (disorder)");
```

Comment:
you misspelled Intolerance here. Not a big deal, but fix if you want.

Reply:
ok

Reply:
corrected